### ISSUE TITLE
Create base_model.py

### ISSUE THREAD

**Issue by KazuCook** [Supika-PROJECT/DL-ImageLab#8]
## Goal
==> State requested to become closed.
base_model : 様々なタスクの親クラスを実装
                      - 各タスクでも扱う汎用的な機能を主として実装
                                            
sr_model : SRモデルの親クラスを実装
                 - SR分野において普遍的な機能を実装する
                 
## Motivation
==> Why do you build issue ?
このIssue下に実装する機能を記載していく
どういう経緯で機能を実装したのか / メンバーとの議論 などの情報を保存する役割を担う

## Tasks
==> What do you try to do ?
- [] BasicSRを参考にbase_model.py / sr_model.pyを実装
- [] Compression.aiを参考に上記モデルの機能を修正 or に機能を追加

## Schedule
==> Schedule until this issue becomes closed.
4/2にfirstリリースを目標に

## Reference
==> In the case of referenced URL
https://github.com/xinntao/BasicSR
https://github.com/InterDigitalInc/CompressAI

**Comment by KazuCook** [comment]
## BaseModel Feature
- model_to_deivce
  Single GPU と DataParallel の二パターンを実装
  後日、Distributedパターンも実装
- print / load / save network
- feature about learning rate 
  まずは、set_lr, update_lr, get_cur_lrを実装
- setup schedulers e.g. MulitiStepLR or CosineAnnealing
   schedulersの詳細なコードはlr_scheduler.py内に
- save_training_state about cur_Ep, cur_iter, optimizers, schedulers
